sdk/samples/iot/paho_iot_hub_pnp_component_sample.c

```diff
@@ -341,13 +331,10 @@ static void create_and_configure_mqtt_client(void)
 static void connect_mqtt_client_to_iot_hub(void)
 {
   // Get the MQTT client username.
-  int rc = az_iot_pnp_client_get_user_name(
-      &pnp_client, mqtt_client_username_buffer, sizeof(mqtt_client_username_buffer), NULL);
-  if (az_result_failed(rc))
-  {
-    IOT_SAMPLE_LOG_ERROR("Failed to get MQTT client username: az_result return code 0x%08x.", rc);
-    exit(rc);
-  }
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(
```

Comment:
Similar to above, a reason for keeping as is here is because it matches the other samples PLUS it mimics the log and exit for the MQTTClient_connect call below.  Your call.